src/clr-angular/forms/combobox/filter-highlight.directive.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import { Directive, ElementRef, OnDestroy, AfterViewInit, Inject, PLATFORM_ID } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { OptionSelectionService } from './providers/option-selection.service';
+import { isPlatformBrowser } from '@angular/common';
+
+@Directive({ selector: '[clrFilterHighlight]' })
+export class ClrFilterHighlight<T> implements AfterViewInit, OnDestroy {
+  private subscriptions: Subscription[] = [];
+  private initialHtml: string;
+  private filter = '';
+  constructor(
+    private element: ElementRef,
+    private optionSelectionService: OptionSelectionService<T>,
+    @Inject(PLATFORM_ID) private platformId: Record<string, any>
+  ) {}
+
+  ngAfterViewInit() {
+    if (isPlatformBrowser(this.platformId)) {
+      this.initialHtml = this.element.nativeElement.innerHTML;
+      this.subscriptions.push(
+        this.optionSelectionService.inputChanged.subscribe(filter => {
+          this.filter = filter;
+          this.findMatches();
+        })
+      );
+    }
+  }
+
+  private sanitizeForRegexp(value: string): string {
+    if (!value) {
+      return value;
+    }
+    // We may want to replace this with a more common sanitization, if we find one
+    return value.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+  }
+
+  private findMatches() {
+    if (!this.filter) {
+      return;
+    }
+    const regex = new RegExp(`(${this.sanitizeForRegexp(this.filter)})`, 'gi');
+    if (this.filter && this.initialHtml.match(regex)) {
+      this.element.nativeElement.innerHTML = this.initialHtml.replace(regex, `<b>$1</b>`);
```

Comment:
We should not be doing this, as it has a potential to be a vector for hijacking a script if we are doing something wrong with our sanitation and DOM manipulation.

If we just want to bold text, we should just add a CSS class that sets the font-weight.

Reply:
We want to be highlighting a substring/s of the whole string. That's the problem. 
Like "**Am**erican S**am**oa" (for search string "am")
We have to drop the whole highlighting scenario if we think it's too risky. I don't mind, it will make the API simpler too. But we have it in the current design specs.

Reply:
Let's review this together sometime this week and make sure we can do this correctly.